src/Test.EndToEnd.Baselining/Extensions/SarifLogExtensions.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Microsoft.CodeAnalysis.Sarif;
+
+namespace SarifBaseline.Extensions
+{
+    public enum BaselineFilteringMode
+    {
+        None = 0,
+        ToIncludedArtifacts
+    }
+
+    public static class SarifLogExtensions
+    {
+        public const string BaselineFilteringPropertyName = "BaselineFiltering";
+
+        public static BaselineFilteringMode GetBaselineFilteringMode(this SarifLog log)
+        {
+            if (log == null) { throw new ArgumentNullException("log"); }
+
+            BaselineFilteringMode result;
+
+            if (log.TryGetProperty<string>(SarifLogExtensions.BaselineFilteringPropertyName, out string modeString))
+            {
+                if (Enum.TryParse<BaselineFilteringMode>(modeString, out result))
+                {
+                    return result;
+                }
+                else
+                {
+                    throw new NotImplementedException($"BaselineFilteringMode '{modeString}' unknown.");
+                }
+            }
+
+            return BaselineFilteringMode.None;
+        }
+
+        public static void SetBaselineFilteringMode(this SarifLog log, BaselineFilteringMode mode)
+        {
+            if (log == null) { throw new ArgumentNullException("log"); }
+            log.SetProperty<string>(SarifLogExtensions.BaselineFilteringPropertyName, mode.ToString());
+        }
+
+        public static IEnumerable<Run> EnumerateRuns(this SarifLog log)
+        {
+            if (log?.Runs != null)
+            {
+                foreach (Run run in log.Runs)
+                {
+                    yield return run;
```

Comment:
>run [](start = 33, length = 3)

In contrast to the novel I wrote about enumerating `artifactLocations`s, this is fine -- this is the only place you'll find a `run`.